Support multi-config variables for the scons generator

olivren opened the pull request:
Just like the cmake and txt generators, the scons generator now
supports so called "multi-config variables".

name = mylib
def package_info(self):
    self.cpp_info.regex.libs = ["myregexlib1", "myregexlib2"]
    self.cpp_info.filesystem.libs = ["myfilesystemlib"]

From this snippet, the generator will create in SConscript_conan the keys:
- conan
- conan:regex
- conan:filesystem
- mylib
- mylib:regex
- mylib:filesystem

All the keys containing a colon are the new generated ones.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/conan-io/conan?pullRequest=1349) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/conan-io/conan?pullRequest=1349) before we can accept your contribution.<br/><hr/>**orenaud** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/conan-io/conan?pullRequest=1349) it.</sub>

memsharded commented:
Nice one! Thanks very much for contributing.

Argh, I just realized there are no docs for SCons: http://conanio.readthedocs.io/en/latest/reference/generators/scons.html :( :(

Created one: https://github.com/conan-io/docs/issues/218

Does Scons now support multi-config debug/release configuration?

memsharded commented:
Also, it seems you are using a different email to sign your commits rather than the github one. Could you please do so to be able to sign the CLA? Thanks very much!

olivren commented:
I did not pay attention, but the email I used for this commit is my professional one. I added this address to my GH profile and signed the CLA, I think it's fine now, is it not?

By the way, I'd like to thank you and all other contributors for this nice package manager. We are deploying it at work to manage the ~25 3rd party libs of our project, and so far I find the design of the tool excellent. Our project is built with SCons on Windows (VS2017) and Linux (GCC6).

Our goal with this PR is to be able to split some libraries into modules: split Qt into QtCore and QtGui, and split Boost into Filesystem and Regex. From what I understand of the "Multi configuration packages" doc page, this is the same mechanism that allows for multi-config debug/release configuration. Not sure, though.

memsharded commented:
Hi @olivren 

Thanks for updating the email, it is perfect now.
Thanks also very much for your compliments. Very happy that you find conan useful! :) :)

About splitting, Boost and Qt, that could be harder than it seems. Would you like to join a slack team to discuss details? Please write us an email to info@conan.io
